### DISCUSSION TITLE
Multibooking discounts

### DISCUSSION THREAD

**Discussion by Sp0run** [liedekef/events-made-easy#321] [Q&A] [answered]
Is it possible to apply a discount to a multibooking if they have booked all events in the recurrence?

The simple code I have written is below, my intention is to check the amount of seats booked against the recurrence ID to evaluate if they have booked all the events so that I can apply the discount. For testing purposes I have hardcoded the the event count and then I have attempted to evaluate the seat count. In this scenario the condition is never met because the filter is being called for every seat making the $seats variable equal to 1. Is there a way round this?

`
add_filter('eme_discount_blockbooking', 'tbr_block_booking_discount');
function tbr_block_booking_discount($booking){

	$disc = 0;
	$seats = $booking['booking_seats'];
	
	if($seats == 4){
	    $disc = 1.10;
	}
	
    return $disc;
}
`

I've had a shuffle through the documentation and found eme_eval_multibooking_form_post_filter filter, but I'm not sure if i would be able to apply a discount update this way.

I had a look on the old forum and found this thread https://www.e-dynamics.be/wordpress/forums/topic/multibooking-scenario/ that says it isn't possible, but was a very long time ago.

**Comment by liedekef** [comment, marked as answer]
eme_eval_multibooking_form_post_filter is for evaluation, it won't change anything and is run before the bookings are inserted into the db. So that'll not work. The problem with multibooking discounts is that I wouldn't know for which event to apply the calculated discount to (because in the end the price is per booking for an event).
You could use the regular discount filter (as above), but you'd need to check also the POST output from the submit to see if all your conditions are met, and then based on that return a discount for that particular booking. An example of checking the POST info is found in eme_eval_multibooking_form_post_filter:
```
    foreach ($_POST['bookings'] as $key=>$val) {
        $total_booked+=$_POST['bookings'][$key]['bookedSeats'];
    }
```
$key contains the event id, so in the foreach you could just count the events with a bookedSeats value>0 or so ...

**Reply by Sp0run** [reply]
Excellent, thanks for that, worked a treat! Incidentally and going off topic a little, it there a way of inserting a variable/placeholder into the recurrence name? For example Round 1, Round 2 etc where the number is the placeholder?


```
$disc = 0;
$seats = $booking['booking_seats'];
$event_id = $booking['event_id'];
$total_booked = 0;
$event_count = tbr_get_recurrence_event_count_from_eventid($event_id);	

foreach ($_POST['bookings'] as $key=>$val) {
        $total_booked+=$_POST['bookings'][$key]['bookedSeats'];
 }
	
	if($total_booked == $event_count){
	    $disc = 0.75;
	}
	
    return $disc;
```

Thanks again for your work on this fantastic plugin

**Reply by liedekef** [reply]
#_RECURRENCE_NBR might be what you're looking for.

**Reply by Sp0run** [reply]
Thanks, Franky. I've tried putting that as the event/recurrence name but on the event it just shows the placeholder as text rather than converting it into a number. I'm guessing that the placeholder cannot be used in the event/recurrence name?

**Reply by liedekef** [reply]
The event name doesn't support placeholders, but you can change the layout of the single event or the event listing to your liking and add it there.

**Reply by Sp0run** [reply]
No problems, I'll have a play with it and see what I can do that works. Thanks again.
